### PR TITLE
ARROW-6329: [Format] Add a padding for Flatbuffer alignment, use 8-byte EOS

### PR DESCRIPTION
**Pull request by emkornfield**
Based on discussion on the mailing list.

### CODE REVIEW

File: docs/source/format/IPC.rst [hunk 1]
```diff
@@ -24,13 +24,16 @@ Encapsulated message format
 Data components in the stream and file formats are represented as encapsulated
 *messages* consisting of:
 
+* A 32-bit continuation indicator (0xFFFFFFFF). This allows flatbuffer bytes
+  to begin on an 8-byte boundary.  This was added as of release 0.15.0.
 * A length prefix indicating the metadata size
```
Comment (pitrou) [inline docs/source/format/IPC.rst]: Change this to "A 32-bit length prefix"?
(also, where is endianness documented?)

**Review by pitrou — COMMENTED** [review]
Just a suggestion.

File: docs/source/format/IPC.rst (hunk 1 above)
Comment (emkornfield) [reply docs/source/format/IPC.rst]: Changed.  Also clarified it is little endian.

**Comment by wesm** [comment]
I added some additional clarification. I'm going to start a vote on the mailing list

File: docs/source/format/IPC.rst [hunk 2]
```diff
@@ -79,14 +83,14 @@ in a ``RecordBatch`` it should be defined in a ``DictionaryBatch``. ::
...
+    <EOS [optional]: 0x00000000>
 
 When a stream reader implementation is reading a stream, after each message, it
 may read the next 4 bytes to know how large the message metadata that follows
 is. Once the message flatbuffer is read, you can then read the message body.
 
-The stream writer can signal end-of-stream (EOS) either by writing a 0 length
-as an ``int32`` or simply closing the stream interface.
+The stream writer can signal end-of-stream (EOS) either by writing the four
+bytes `0x00000000` or closing the stream interface.
```
Comment (pitrou) [inline docs/source/format/IPC.rst]: Could we take the opportunity to turn this into a 8-bytes marker?

**Pull request closed without merging**